CONTRIBUTING.md

```diff
@@ -0,0 +1,96 @@
+small--is greatly appreciated. In this file, we'll outline how you can
+contribute to the project.
+
+Xournal++ development primarily occurs on [GitHub at the xournalpp/xournalpp
+repository](https://github.com/xournalpp/xournalpp). As a contributor, you
+probably have a particular bug or feature that you are interested in working on.
+Before you start, you should first look in the [issue
+tracker](https://github.com/xournalpp/xournalpp/issues) to see if anyone has
+reported your bug or has a similar idea for a feature. If not, you should first
+submit a new issue detailing what you are about to do. This will allow you to
+get feedback and ensure that you do not end up duplicating work. You can obtain
+additional help by contacting community members through one of our [official
+communication channels](https://xournalpp.github.io/community/help/).
+
+## Contributing translations
+
+If you would like to contribute translations, you can submit improvements to
+[our project on Crowdin](https://crowdin.com/project/xournalpp).
+
+## Contributing code changes with pull requests (PRs)
+
+To make a code change, you'll need to first fork the repository on GitHub, if
+you haven't already. To do so, click the "Fork" button at the top right corner
+of the repository's web page.
+
+We assume that you are already familiar with Git, which we use to track changes.
+An excellent resource for Git is the _Pro Git_ book by Scott Chacon and Ben
+Straub, which is available online for free
+[here](https://git-scm.com/book/en/v2).
+
+You should try to first clone your fork and then compile it manually. The
+instructions for compiling Xournal++ on your operating system can be found at
+[`LinuxBuild.md`](readme/LinuxBuild.md), [`MacBuild.md`](readme/MacBuild.md),
+and [`WindowsBuild.md`](readme/WindowsBuild.md). Once you are set up, you will
+be ready to make code changes.
+
+### Code conventions and guidelines
+
+Here is a quick list of guidelines that we follow when working on the Xournal++
+codebase.
+
+* Xournal++ is a mixture of C++ and C code. This is unavoidable because we rely on
+  many C libraries (e.g. GTK). Prefer C++ when possible, however.
+* Although we do not have a strong opinion on code style, we use `clang-format`
+  to enforce a _consistent_ code style. Feel free to write your code in whatever
+  style you prefer, as long as you run `clang-format` to format your code
+  afterwards.
+* Please accompany your code changes with documentation.
+* Follow modern C++ best practices as listed in the [C++ Core
+  Guidelines](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines). In
+  particular, make variables `const` by default, always use smart pointers
+  instead of new/delete, and prefer `std::optional` over null pointers.
+* When working with code that interfaces with C libraries (GLib and GTK in
+  particular), you should be extra careful with reference counting and memory
+  management.
```

Comment:
```suggestion
  management.
  It might be helpfull, to create own memory managing RAII structures. This will also reduce the time others must spend to verify your code, because it is correct by design.
```